server2/Arcadia.Assistant/Arcadia.Assistant.AppCenterBuilds/UpdateMobileBuildHelper.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Arcadia.Assistant.AppCenterBuilds.Contracts.AppCenter;
+using Arcadia.Assistant.MobileBuild.Contracts.Interfaces;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Arcadia.Assistant.AppCenterBuilds
+{
+    internal class UpdateMobileBuildHelper
+    {
+        private readonly string buildUrl;
+        private readonly string downloadUrlTemplate;
+        private readonly string apiKey;
+
+        public UpdateMobileBuildHelper(string buildUrl, string downloadUrlTemplate, string apiKey)
+        {
+            this.buildUrl = buildUrl;
+            this.downloadUrlTemplate = downloadUrlTemplate;
+            this.apiKey = apiKey;
+        }
+
+        #region public mehods
+
+        public async Task<bool> CheckAndupdateMobileBuild(IHttpClientFactory httpClientFactory, IMobileBuildActor mobileBuildActor, CancellationToken cancellationToken, Action<string> logStore)
```

Comment:
why boolean? Why Helper? This is an update process / that consists of multiple steps, split as such.

Reply:
Skip this code for future redesign.